edx_rest_framework_extensions/tests/test_permissions.py

```diff
@@ -155,3 +182,176 @@ def test_has_permission(self, enforce_scopes, authentication_class, jwt_filters,
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=JwtHasScopeLog),
+
+        # **** Unenforced ****
+        # unrestricted
+        JwtTestCase(is_enforced=False, is_restricted=False, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=False, is_restricted=False, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+
+        # restricted
+        JwtTestCase(is_enforced=False, is_restricted=True, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=False, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+    )
+    @ddt.unpack
+    def test_jwt_cases(self, is_enforced, is_restricted, is_user_in_url, expected_response, expected_log, mock_log, waffle_mock):
+        waffle_mock.return_value = is_enforced
+        user = self._create_user()
+
+        auth_header = self._create_jwt_header(user, is_restricted=is_restricted)
+        request = self._create_request(
+            username_in_url=user.username if is_user_in_url else None,
+            auth_header=auth_header,
+        )
+
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, expected_response)
+        self._assert_log(mock_log, expected_log)
+
+    JwtRestrictedTestCase = namedtuple('JwtRestrictedTestCase', 'scopes, filters, expected_response, expected_log')
+
+    @patch('edx_rest_framework_extensions.permissions.waffle.switch_is_active')
+    @patch('edx_rest_framework_extensions.permissions.log')
+    @ddt.data(
+        JwtRestrictedTestCase(scopes=['required_scope'], filters=['content_org:some_org'], expected_response=200, expected_log=JwtIsRestrictedLog),
+        JwtRestrictedTestCase(scopes=['required_scope', 'another_scope'], filters=['content_org:some_org'], expected_response=200, expected_log=JwtIsRestrictedLog),
+        JwtRestrictedTestCase(scopes=['required_scope'], filters=['content_org:some_org', 'some:other'], expected_response=200, expected_log=JwtIsRestrictedLog),
+
+        JwtRestrictedTestCase(scopes=['required_scope'], filters=['content_org:another_org'], expected_response=403, expected_log=JwtOrgFilterLog),
+        JwtRestrictedTestCase(scopes=['required_scope'], filters=[], expected_response=403, expected_log=JwtOrgFilterLog),
+
+        JwtRestrictedTestCase(scopes=[], filters=['content_org:some_org'], expected_response=403, expected_log=JwtHasScopeLog),
+        JwtRestrictedTestCase(scopes=['another_scope'], filters=['content_org:some_org'], expected_response=403, expected_log=JwtHasScopeLog),
+    )
+    @ddt.unpack
+    def test_jwt_enforced_restricted(self, scopes, filters, expected_response, expected_log, mock_log, waffle_mock):
```

Comment:
I thought this should be `test_jwt_with_scopes_and_filters`?  But, is it `test_jwt_org_filters`?  Do we have consistent coverage for user filter?  I don't think it is quite the same, because you didn't mess with scopes.  Do you need to mess with scopes in this test, or could required_scopes always be there and be removed from the params?

Reply:
do this.

Reply:
done.